pkg/operator/ceph/pool/radosnamespace/controller.go

```diff
@@ -0,0 +1,312 @@
+}
+
+// ReconcileCephBlockPoolRadosNamespace reconciles a CephBlockPoolRadosNamespace object
+type ReconcileCephBlockPoolRadosNamespace struct {
+	client           client.Client
+	scheme           *runtime.Scheme
+	context          *clusterd.Context
+	clusterInfo      *cephclient.ClusterInfo
+	opManagerContext context.Context
+}
+
+// Add creates a new CephBlockPoolRadosNamespace Controller and adds it to the
+// Manager. The Manager will set fields on the Controller and Start it when the
+// Manager is Started.
+func Add(mgr manager.Manager, context *clusterd.Context, opManagerContext context.Context, opConfig opcontroller.OperatorConfig) error {
+	return add(mgr, newReconciler(mgr, context, opManagerContext))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager, context *clusterd.Context, opManagerContext context.Context) reconcile.Reconciler {
+	return &ReconcileCephBlockPoolRadosNamespace{
+		client:           mgr.GetClient(),
+		scheme:           mgr.GetScheme(),
+		context:          context,
+		opManagerContext: opManagerContext,
+	}
+}
+
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+	logger.Info("successfully started")
+
+	// Watch for changes on the CephBlockPoolRadosNamespace CRD object
+	err = c.Watch(&source.Kind{Type: &cephv1.CephBlockPoolRadosNamespace{TypeMeta: controllerTypeMeta}}, &handler.EnqueueRequestForObject{}, opcontroller.WatchControllerPredicate())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Reconcile reads that state of the cluster for a CephBlockPoolRadosNamespace
+// object and makes changes based on the state read and what is in the
+// CephBlockPoolRadosNamespace.Spec The Controller will requeue the Request to be
+// processed again if the returned error is non-nil or Result.Requeue is true,
+// otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileCephBlockPoolRadosNamespace) Reconcile(context context.Context, request reconcile.Request) (reconcile.Result, error) {
+	// workaround because the rook logging mechanism is not compatible with the controller-runtime logging interface
+	reconcileResponse, err := r.reconcile(request)
+	if err != nil {
+		logger.Errorf("failed to reconcile %v", err)
+	}
+
+	return reconcileResponse, err
+}
+
+func (r *ReconcileCephBlockPoolRadosNamespace) reconcile(request reconcile.Request) (reconcile.Result, error) {
+	// Fetch the CephBlockPoolRadosNamespace instance
+	cephBlockPoolRadosNamespace := &cephv1.CephBlockPoolRadosNamespace{}
+	err := r.client.Get(r.opManagerContext, request.NamespacedName, cephBlockPoolRadosNamespace)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			logger.Debug("cephBlockPoolRadosNamespace resource not found. Ignoring since object must be deleted.")
```

Comment:
The rook operator can operate on multiple resources in multiple namespaces. All info/debug messages must list the namespaced name so context is clear.